libgraph/src/analytics/k_shortest_paths/ksssp.cpp

```diff
@@ -0,0 +1,415 @@
+      edge_data[e] = graph.template GetEdgeData<EdgeWeight>(e);
+    }
+  });
+
+  katana::StatTimer execTime("kSSSP");
+  execTime.start();
+
+  katana::InsertBag<std::pair<Weight, Path*>> paths;
+  katana::InsertBag<Path*> path_pointers;
+
+  bool reachable = true;
+
+  switch (algo_reachability) {
+  case async:
+    reachable = CheckReachabilityAsync<BFSUpdateRequest>(
+        &graph, source, BFSReqPushWrap(), BFSOutEdgeRangeFn{&graph},
+        report_node);
+    break;
+  case syncLevel:
+    reachable = CheckReachabilitySync(&graph, source, report_node);
+    break;
+  default:
+    std::abort();
+  }
+
+  PathAlloc path_alloc;
+
+  if (reachable) {
+    switch (plan.algorithm()) {
+    case kSsspPlan::kDeltaTile:
+      DeltaStepAlgo<kSSSPSrcEdgeTile, OBIM>(
+          &edge_data, &graph, source, kSSSPSrcEdgeTilePushWrap{&graph},
+          kSSSPTileRangeFn(), &paths, &path_pointers, path_alloc, report_node,
+          num_paths, step_shift);
+      break;
+    case kSsspPlan::kDeltaStep:
+      DeltaStepAlgo<kSSSPUpdateRequest, OBIM>(
+          &edge_data, &graph, source, kSSSPReqPushWrap(),
+          kSSSPOutEdgeRangeFn{&graph}, &paths, &path_pointers, path_alloc,
+          report_node, num_paths, step_shift);
+      break;
+    case kSsspPlan::kDeltaStepBarrier:
+      katana::gInfo("Using OBIM with barrier\n");
+      DeltaStepAlgo<kSSSPUpdateRequest, OBIM_Barrier>(
+          &edge_data, &graph, source, kSSSPReqPushWrap(),
+          kSSSPOutEdgeRangeFn{&graph}, &paths, &path_pointers, path_alloc,
+          report_node, num_paths, step_shift);
+      break;
+
+    default:
+      return katana::ErrorCode::InvalidArgument;
+    }
+  }
+
+  execTime.stop();
+  page_alloc.Report();
+
+  if (reachable) {
+    std::multimap<uint32_t, Path*> paths_map;
+
+    for (auto pair : paths) {
+      paths_map.insert(std::make_pair(pair.first, pair.second));
+    }
+
+    katana::gPrint("Node ", report, " has these k paths:\n");
+
+    uint32_t num =
+        (paths_map.size() > num_paths) ? num_paths : paths_map.size();
+
+    auto it_report = paths_map.begin();
+
+    for (uint32_t iter = 0; iter < num; iter++) {
+      const Path* path = it_report->second;
+      PrintPath(path);
+      katana::gPrint(" ", report, "\n");
+      katana::gPrint("Weight: ", it_report->first, "\n");
+      it_report++;
+    }
+
+    katana::do_all(katana::iterate(path_pointers), [&](Path* p) {
+      path_alloc.DeletePath(p);
+    });
+  }
+
+  return katana::ResultSuccess();
+}
+
+katana::Result<void>
+katana::analytics::Ksp(
+    katana::PropertyGraph* pg, unsigned int start_node,
+    unsigned int report_node, katana::TxnContext* txn_ctx,
+    AlgoReachability algo_reachability, unsigned int num_paths,
+    unsigned int step_shift, kSsspPlan plan) {
+  auto result =
+      ConstructNodeProperties<std::tuple<NodeCount, NodeMax>>(pg, txn_ctx);
+  if (!result) {
+    return result.error();
+  }
+
+  using GraphTy = katana::TypedPropertyGraph<
```

Comment:
this part still need to be changed to use Default and undirected views. Check the k-core app https://github.com/uditagar/katana/blob/a0ab4ac12da2e6d20add5a12151a972a4ee894e7/libgraph/src/analytics/k_core/k_core.cpp#L240 lines 240-255. See how it is done here.